test/Kestrel.Core.Tests/HPackEncoderTests.cs

```diff
@@ -10,6 +10,40 @@ namespace Microsoft.AspNetCore.Server.Kestrel.Core.Tests
 {
     public class HPackEncoderTests
     {
+        [Fact]
+        public void LowercaseHeaderNamesOnlyWhenEncodingHeaders()
+        {
+
+            var encoder = new HPackEncoder();
+            var headers = new[]
+            {
+                new KeyValuePair<string, string>("CustomHeader", "CustomValue"),
```

Comment:
Do you plan to add test cases for the the cases that currently aren't properly handled?

Reply:
Not for this, I filed #2684 separately to track that.

Reply:
What cases aren't properly handled?